File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/SiteSettingsPage.java
package org.wordpress.android.e2e.pages;

import androidx.test.espresso.ViewInteraction;

import org.wordpress.android.R;

import static androidx.test.espresso.Espresso.onData;
import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.scrollTo;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.PreferenceMatchers.withTitle;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.wordpress.android.support.WPSupportUtils.ensureSwitchPreferenceIsChecked;

public class SiteSettingsPage {
    private static ViewInteraction settings = onView(withText(R.string.site_settings_title_title));

    public SiteSettingsPage() {
        settings.check(matches(isDisplayed()));
    }

    public void setEditorToClassic() {
        onData(withTitle(R.string.site_settings_gutenberg_default_for_new_posts))
                .perform(scrollTo())
                .perform(ensureSwitchPreferenceIsChecked(false));
    }

    public void setEditorToGutenberg() {
        onData(withTitle(R.string.site_settings_gutenberg_default_for_new_posts))
                .perform(scrollTo())
                .perform(ensureSwitchPreferenceIsChecked(true));
    }
}

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/SiteSettingsPage.kt
package org.wordpress.android.e2e.pages

import androidx.test.espresso.Espresso
import androidx.test.espresso.assertion.ViewAssertions
import androidx.test.espresso.matcher.PreferenceMatchers
import androidx.test.espresso.matcher.ViewMatchers
import org.wordpress.android.R
import org.wordpress.android.support.BetterScrollToAction.Companion.scrollTo
import org.wordpress.android.support.WPSupportUtils

class SiteSettingsPage {
    init {
        settings.check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
    }

    fun setEditorToClassic() {
        Espresso.onData(PreferenceMatchers.withTitle(R.string.site_settings_gutenberg_default_for_new_posts))
            .perform(scrollTo())
            .perform(WPSupportUtils.ensureSwitchPreferenceIsChecked(false))
    }

    fun setEditorToGutenberg() {
        Espresso.onData(PreferenceMatchers.withTitle(R.string.site_settings_gutenberg_default_for_new_posts))
            .perform(scrollTo())
            .perform(WPSupportUtils.ensureSwitchPreferenceIsChecked(true))
    }

    companion object {
        private val settings =
            Espresso.onView(ViewMatchers.withText(R.string.site_settings_title_title))
    }
}
